users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -49,6 +49,9 @@ def add_specaug_layer(
                 freq_mask_max_proportion (float): The maximum proportion of the frequency axis that can be masked. Default is 1.
                 max_time_num_seq_len_divisor (int):
                     The divisor for the sequence length to determine the maximum number of time masks. Default is 0.7.
+                mask_based_on_variance (bool): Whether to mask based on variance or not. Default is False.
+                variance_factor (float): The factor that determines the probability of masking a feature based on variance. Default is 0.5.
+                total_max_number_masks (int): The total maximum number of masks to be applied if variance-based masking is used. Default is 50.
```

Comment:
Since variance based masking does not use a number of different sizes of masks and instead is masking single features based on variance, I need a parameter for the maximum number of masked features. I could calculate it using the existing ones (max_feature_num * max_feature) but that would overload the variables and make the behavior very hard to understand.

Reply:
Alright, but then please make the name reflect that this is only for the filter content based masking.